redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/termination/AbstractRedbeamsTerminationAction.java

```diff
@@ -64,29 +64,30 @@ protected void doExecute(RedbeamsContext context, P payload, Map<Object, Object>
 
     @Override
     protected RedbeamsContext createFlowContext(FlowParameters flowParameters,
-        StateContext<RedbeamsTerminationState, RedbeamsTerminationEvent> stateContext, P payload) {
+            StateContext<RedbeamsTerminationState, RedbeamsTerminationEvent> stateContext, P payload) {
         Optional<DBStack> optionalDBStack = dbStackService.findById(payload.getResourceId());
-
         CloudContext cloudContext = null;
         CloudCredential cloudCredential = null;
         DatabaseStack databaseStack = null;
         DBStack dbStack = null;
-        dbStack = optionalDBStack.get();
-        MDCBuilder.buildMdcContext(dbStack);
-        Location location = location(region(dbStack.getRegion()), availabilityZone(dbStack.getAvailabilityZone()));
-        String userName = dbStack.getOwnerCrn().getUserId();
-        String accountId = dbStack.getOwnerCrn().getAccountId();
-        cloudContext = new CloudContext(dbStack.getId(), dbStack.getName(), dbStack.getResourceCrn().toString(), dbStack.getCloudPlatform(),
-                dbStack.getPlatformVariant(), location, userName, accountId);
-
         if (optionalDBStack.isPresent()) {
```

Comment:
In other termination flow steps if DBStack is not present then it will cause NPE. I think it might make sense to propagate the Optional to those places as well and fail with a custom exception when there is no DBStack.